docs/import/manage_disaster.js

```diff
@@ -61,48 +61,90 @@ let scoreBoundsMap;
 
 const scoreCoordinatesPath = Object.freeze([scoreCoordinatesAttribute]);
 
+const kickOffText = 'Kick off Data Processing (will take a while!)';
+const optionalWarningPrefix = '; warning: created asset will be missing ';
+
 /**
- * Checks that all fields that can be entered by the user have a non-empty
- * value. Does not check that assets actually exist, are of valid type, etc. If
- * all validation succeeds, enables kick-off button, otherwise disables and
- * changes button text to say what is missing.
+ * Checks that all mandatory fields that can be entered by the user have
+ * non-empty values. Does not check that assets actually exist, are of valid
+ * type, etc. If all validation succeeds, enables kick-off button, otherwise
+ * disables and changes button text to say what is missing.
+ *
+ * Some fields (Income, SVI) are optional. If they are missing, a separate
+ * message is displayed on the button, but it can still be enabled. If it is
+ * enabled, it is yellowed a bit to indicate the missing optional assets.
+ * The buildings asset is optional if the damage asset is not present, but is
+ * required if the damage asset is present.
  */
 function validateUserFields() {
   const states = disasterData.get(getDisaster()).states;
   const missingItems = [];
-  for (const scoreAssetType of scoreAssetTypes) {
+  const damageAssetPresent = !!$('#damage-asset-select').val();
+
+  for (const {idStem, displayName} of scoreAssetTypes) {
     const missingForType = [];
     for (const state of states) {
-      if (!$('#select-' + assetSelectionRowPrefix + scoreAssetType[0] + '-' +
-             state)
+      if (!$('#select-' + assetSelectionRowPrefix + idStem + '-' + state)
                .val()) {
         missingForType.push(state);
       }
     }
     if (missingForType.length) {
-      missingItems.push(
-          scoreAssetType[2] +
-          (states.length > 1 ? ' [' + missingForType.join(', ') + ']' : ''));
+      const missingItem = [displayName];
+      if (states.length > 1) {
+        missingItem.push('[' + missingForType.join(', ') + ']');
+      }
+      missingItems.push(missingItem);
     }
   }
-  const hasDamage = $('#damage-asset-select').val() ||
-      getElementFromPath(scoreCoordinatesPath);
+  const hasDamage =
+      damageAssetPresent || getElementFromPath(scoreCoordinatesPath);
   let message = '';
+  let optionalMessage = '';
   if (missingItems.length) {
-    message = 'Missing asset(s): ' + missingItems.join(', ');
+    for (const missingItem of missingItems) {
+      const optionalBuildings =
```

Comment:
This logic is starting to get quite hairy (though looks good in the button!) If we have time (lol) maybe we can think about more visual messaging through icons or something.

Reply:
Agree it's getting hairy, wish I had an easier way. Can think about a refactoring soon.